haskell-decl-scan.el

```diff
@@ -382,18 +377,17 @@ there."
     nil))
 
 (defun haskell-ds-line-commented-p ()
-  "Tests if all characters from `point' to `end-of-line' pass
-`haskell-ds-comment-p'"
-  (let ((r t))
-    (while (and r (not (eolp)))
-      (if (not (haskell-ds-comment-p))
-          (setq r nil))
-      (forward-char))
-    r))
+  "Test if all characters from `point' to `end-of-line' pass `haskell-ds-comment-p'."
```

Comment:
Use `(parse-partial-sexp (point) (point-max) nil nil (syntax-ppss) 'syntax-table)` for this purpose.